pkg/controller/middleware/sessions.go

```diff
@@ -65,6 +68,17 @@ func buildHandler(store sessions.Store, name string, h *render.Renderer) func(ht
 				// have a session.
 				session, _ = store.New(r, name)
 			}
+			var authSession *sessions.Session
+			if authProvider != nil {
+				authSession, err = store.Get(r, authSessionName)
+				if err != nil {
+					logger.Errorw("failed to get auth session", "error", err)
+					authSession, _ = store.New(r, name)
+				}
+				if err = authProvider.JoinAuthCookie(session, authSession); err != nil {
```

Comment:
I think you can do the cookie splitting and joining here, and avoid needing to hang this off of the `auth.Provider` entirely:

On save:

- Extract and remove fields from session.Values and put into a new session that has the same store. These are public fields so we can easily write a Clone function or something.
- Save both sessions (or 3 or 4 if we wanted in the future)

On load:

- Download all cookies and merge into a single session.Values map.

Reply:
splitting all values is unnecessary overhead for network and computation. I think it's best to do this surgical fix.

alternatively - I could have the session provider take in a list of keys that should be split to own cookies.

how does that sound?

Reply:
Sorry - I didn't mean _every_ key. We know that the firebase key is a problem. We can explicitly pull that one out into its own cookie and explicitly merge it back in.

Reply:
TL;DR - I'd prefer to keep this out of the `authProvider` if possible.

Reply:
simplified.